docs/development/pigs/pig-026.rst

```diff
@@ -0,0 +1,328 @@
+.. include:: ../../references.txt
+
+.. _pig-026:
+
+********************************
+PIG 26 - Model Priors API 
+********************************
+
+* Author: Noah Biederbeck, Katrin Streil
+* Created: ``2023-03-13``
+* Accepted:
+* Status: draft
+* Discussion: `#4381`_
+
+Abstract
+========
+
+This PIG is intended to introduce priors on models or on parameters that are evaluated during fitting.
+
+Motivation
+==========
+
+Using priors on models or on parameters is the next step in full-fledged statistical analyses. 
+A prior can incorporate the analysers' knowledge of the topic at hand or information about
+the estimated IRFs systematics provided by the corresponding experiment, and yield more realistic and trustworthy results. 
+The proposed formalism also includes the application of nuisance parameters and regularization.
+
+Use cases
+=========
+
+In the past priors were a regularly requested feature or the solution to a problem.
+See the following issues and PRs of Gammapy:
+
+- Add sigma v estimator functionality for dark matter use case `#2075`_
+- Background systematics as a nuisance parameter `#3955`_
+- Support unfolding methods for spectral flux points `#4122`_
+- Add PiecewiseNormSpatialModel `#4208`_
+
+Implementation
+==============
+
+The following implementation draft is compatible with the current API,
+where the models are set via Dataset.model and the Fit is run via Fit.run(datasets).
+
+Prior (abstract) base class:
+--------------
+
+.. code:: python 
+
+  class Prior():
+  
+     def __call__(self, base):
+     
+        if isinstance(base, Parameter):
+            return self.evaluate(base.value)      
+        else: # model-like instance
+            return self.evaluate(base.parameters.value)  
+         
+
+Future additional properties of the ``Prior`` classes: 
+
+- Fittable parameters
+- Write/read from/to a yaml file, ideally also when the corresponding model is written/read (see serialisation example below)
+- Prior registry system
+- Prior subclass use cases (i.e. for correlated nuisance parameters with correlation matrix as a parameter)
+
+Exemplary additional prior subclasses:
+--------------
+.. code:: python
+
+  class GaussianPrior(Prior):
+  
+      """Gaussian Prior with mu and sigma set on a single parameter
+      """
+      
+      def __init__(self, mu=0, sigma=1):
+          self.mu = mu
+          self.sigma = sigma
+
+      def evaluate(self, value):
```

Comment:
Yes, I agree that the `.evaluate()` method should be static if possible. This will allows us to easier move to other tensor libraries later...The stateful call can happen in `Prior.__call__`